src/azure-cli/azure/cli/command_modules/apim/commands.py

```diff
@@ -24,3 +29,10 @@ def load_command_table(self, _):
         g.custom_command('check-name', 'check_name_availability')
         g.custom_command('backup', 'apim_backup', supports_no_wait=True)
         g.custom_command('apply-network-updates', 'apim_apply_network_configuration_updates', supports_no_wait=True)
+
+    with self.command_group('apim api', api_sdk, is_preview=True) as g:
+        g.custom_command('create', 'create_apim_api', supports_no_wait=True)
```

Comment:
A `wait` command is also needed if there are commands which supports no wait.

Reply:
I don't get it quite, the only "wait" command I saw in the existing methods was as follows:
```return sdk_no_wait(no_wait, cms.create_or_update, ...)```.

So for:
1. `g.custom_command('create', 'create_apim_api', supports_no_wait=True)`
I've used the ```return sdk_no_wait```

2. `g.generic_update_command('update', custom_func_name='update_apim_api', supports_no_wait=True)`
I've created the method similar to the existing ```az apim update``` method

3. `g.custom_command('delete', 'delete_apim_api', confirmation=True, supports_no_wait=True)`
**I will add** a: ```return sdk_no_wait(no_wait, cms.delete...)```

Do I need to add something on 1. and 2.?

Reply:
I mean we also need to add a `az apim api wait` command. There are so many examples in our code base. See [here ](https://github.com/Azure/azure-cli/blob/4c28536a6db8c8e43b7182d0aaf202ac2b4ab777/src/azure-cli/azure/cli/command_modules/vm/commands.py#L204)for reference.

The reason we need to add `wait` command is that, if the user specify `--no-wait` for some command, he can also use `wait` command to check the provisioning status later.

Reply:
Ah now I get it thanks for the clarification. I've added a wait command to the commands.py.